### ISSUE TITLE
Extend proactive tagging to be able to rename objects

### ISSUE THREAD

**Issue by Tobias-Fischer** [robotology/wysiwyd#38]
During the integration, we talked about having the possibility to re-train the names of objects. For example, the iCub might think an object is called `apple`, but its actual name is `carrot`. This might be due to a) wrongly detected pointing in the training phase or b) `apple` might be a fantasy name of the iCub.

A simple implementation might be to just use the already existing code, but extend it such that objects which do not start with `unknown_object` are considered. Then, we should be able to say `iCub, point the carrot`. Then, iCub responds `I don't know which of these objects is the carrot`. We then point the `apple` (which is actually the `carrot`). The iCub then says: Now I know that is the `carrot`.

A more advanced version would change the name step for step, so rather than one-shot learning we would need to teach the iCub the name of an object over several iterations.

**Comment by maxime-petit** [comment]
For this, there are 2 method in proactive:
- exploreUnknownEntity : does not care I think about the name unknown or not, so you can actually rename an object. It is just that the order (coming from allostatic) should allow that
- searchEntity : in that case, inside we check that "unkown" is part of the current name, so here we will need to change this

It is because the first one is for the robot self exploratory, and the second is the "goal-directed" exploration after an order. 

Basically, we may seen the "renaming" as a 3rd behavior and in that case take care of that independtly. Or we see that as a more "general" way to deal with the first 2 initial behaviors.

Thoughts?

**Comment by Tobias-Fischer** [comment]
@clement-moulin-frier: What's the status on this?

**Comment by clement-moulin-frier** [comment]
I've pushed a first version to the branch [upf/renaming](https://github.com/robotology/wysiwyd/tree/upf/renaming). I've also started to deal with random choice of sentences but can't really go further without being able to test it before.

**Comment by Tobias-Fischer** [comment]
Hi @clement-moulin-frier,
Can you please open a pull request for the upf/renaming branch? This makes it easier to keep track of it. I hope we can test this soon and merge.

Best, Tobi

**Comment by Tobias-Fischer** [comment]
We should also decide whether this is one of the features we want to show in year 2.
@clement-moulin-frier @pattacini @maxime-petit

**Comment by clement-moulin-frier** [comment]
Hi guys,

I'll try and debug the new feature when I'll be back to the lab next week. I will make the pull request at this stage if the behavior is satisfactory.

Clément

**Comment by Tobias-Fischer** [comment]
Any news on this @clement-moulin-frier?

**Comment by clement-moulin-frier** [comment]
We were not able to finish it this week, we'll update next week through a pull request

**Comment by Tobias-Fischer** [comment]
Hi @clement-moulin-frier, did you do further progress on this? I hope the refactoring did not cause too many issues.

**Comment by clement-moulin-frier** [comment]
@jypuigbo is currently re-integrating the code after you pull request merge. It should be fine.

**Comment by jypuigbo** [comment]
see PR #110

**Comment by Tobias-Fischer** [comment]
Should we close this @jypuigbo @clement-moulin-frier @maxime-petit? Don't think this will happen, I think it's a `wontfix`.

**Comment by clement-moulin-frier** [comment]
Yes let's close it for the time being and we'll reopen it if we need the feature for Y3. 
Not complicated to do but there many possible ways and we need to agree on one.
